# Домашнее задание к классу N5

**Цель:** Для матрицы 

$$A = \begin{pmatrix} 
3 & 7 \\
5 & 2 
\end{pmatrix}$$

выполнить сингулярное разложение в python с использованием библиотеки numpy (функция `linalg.svd`). Записать получившуюся матрицу преобразования.

In [17]:
import numpy as np
import numpy.linalg as la

In [18]:
A = np.array([
    [3, 7],
    [5, 2]
])

display(A)

array([[3, 7],
       [5, 2]])

In [19]:
# вычисляем сингулярное разложение
U, s, V = la.svd(A)

In [20]:
# преобразуем вектор диагонали в диагональную матрицу
D = np.diag(s)

In [21]:
# Проверка верности разложения
U @ D @ V

array([[3., 7.],
       [5., 2.]])

In [22]:
display(U)
display(D)
display(V)

array([[-0.85065081, -0.52573111],
       [-0.52573111,  0.85065081]])

array([[8.71337969, 0.        ],
       [0.        , 3.32821489]])

array([[-0.59455781, -0.80405286],
       [ 0.80405286, -0.59455781]])

## Ответ

$$A = \begin{pmatrix} 
3 & 7 \\
5 & 2 
\end{pmatrix} = U \times D \times V = 
\begin{pmatrix} 
-0.85065 & -0.52573 \\
-0.52573 &  0.85065
\end{pmatrix}
\times
\begin{pmatrix} 
8.71337 & 0 \\
0       & 3.32821 
\end{pmatrix}
\times
\begin{pmatrix} 
-0.59455 & -0.80405 \\
 0.80405 & -0.59455
\end{pmatrix}$$